Permission and Access Helper/force-app/main/default/classes/PermissionServiceImpl.cls

```diff
@@ -0,0 +1,264 @@
+    
+ 	public List<ComboboxDataWrapper> getProfile(){
+        return convertProfilesToWrapper(ProfileSelector.newInstance().getAllRecords());
+ 	}
+ 	public List<ComboboxDataWrapper> getPermissionSet(){
+        return convertPermissionSetsToWrapper(PermissionSetSelector.newInstance().getAllRecords());
+ 	}
+    public List<PermissionDataWrapper> getAllPermissionData(){
+        List<FieldPermissions> fieldPermissions = FieldPermissionsSelector.newInstance().getAllRecords();
+        List<ObjectPermissions> objectPermissions = ObjectPermissionSelector.newInstance().getAllRecords();
+        List<PermissionSet> permissionSets = PermissionSetSelector.newInstance().getAllRecords();
+        List<Profile> profiles = ProfileSelector.newInstance().getAllRecords();
+        List<PermissionDataWrapper> dataToReturn = convertAllDataToWrapper(fieldPermissions, objectPermissions, permissionSets, profiles);
+        return dataToReturn;
+    }
+    public List<ComboboxDataWrapper> getObjectOptionsForObjectPermissions(){
+        List<ObjectPermissions> objectPermissions = ObjectPermissionSelector.newInstance().getAllRecords();
+        List<ComboboxDataWrapper> wrappedData = new List<ComboboxDataWrapper>();
+        Set<String> objectsInUse = new Set<String>();
+        for(ObjectPermissions op : objectPermissions){
+            objectsInUse.add(op.SobjectType);
+        }
+        List<String> objectApinameList = new List<String>();
+        objectApinameList.addAll(objectsInUse);
+        objectApinameList.sort();
+        Map<String, SObjectType> objectMap = Schema.getGlobalDescribe();
+        for(String obj : objectApinameList){
+            SObjectType objectFromMap = objectMap.get(obj);
+            String label = obj;
+            if(objectFromMap != null){
+                label = objectFromMap.getDescribe().getLabel();
+            }
+            wrappedData.add(new ComboboxDataWrapper(label, obj));
+        }
+        return wrappedData;
+    }
+    public List<ComboboxDataWrapper> getObjectOptionsForFieldPermissions(){
+        List<FieldPermissions> fieldPermissions = FieldPermissionsSelector.newInstance().getAllRecords();
+        List<ComboboxDataWrapper> wrappedData = new List<ComboboxDataWrapper>();
+        Set<String> objectsInUse = new Set<String>();
+        for(FieldPermissions fp : fieldPermissions){
+            objectsInUse.add(fp.SobjectType);
+        }
+        List<String> objectApinameList = new List<String>();
+        objectApinameList.addAll(objectsInUse);
+        objectApinameList.sort();
+        Map<String, SObjectType> objectMap = Schema.getGlobalDescribe();
+        for(String obj : objectApinameList){
+            SObjectType objectFromMap = objectMap.get(obj);
+            String label = obj;
+            if(objectFromMap != null){
+                label = objectFromMap.getDescribe().getLabel();
+            }
+            wrappedData.add(new ComboboxDataWrapper(label, obj));
```

Comment:
what was the difference when the objectFromMap is null then we create wrapper with same value why?

Reply:
There are some cases in which the object is not fetched in getGlobalDescribe, which were discussed earlier, so for them, we have to add this condition such that if the object is not fetched, it's API name is used as it's label